core/commands/object.go

```diff
@@ -701,6 +702,24 @@ func objectPut(n *core.IpfsNode, input io.Reader, encoding string) (*Object, err
 	case objectEncodingProtobuf:
 		dagnode, err = dag.Decoded(data)
 
+	case objectEncodingXML:
+		node := new(Node)
+		err = xml.Unmarshal(data, node)
+		if err != nil {
+			return nil, err
+		}
+
+		// check that we have data in the Node to add
+		// otherwise we will add the empty object without raising an error
+		if NodeEmpty(node) {
+			return nil, ErrEmptyNode
+		}
```

Comment:
> I looks like its copying the behaviour of the json parsing:

Ah right.

> Is that logic not what we want to be doing now? (I think it would probably be good to allow empty inputs to 'put')

Yeah, i think we do want that. I'll merge this now, and that can be a separate PR.